### PR TITLE
more events

### PR DESCRIPTION
**Pull request by hzjzx**
more events

### CODE REVIEW

**Comment by egoist** [comment]
what do you think of using an event to listen to all editor events? like:

```vue
<template>
     <monaco-editor @editor-event="handleEditorEvent" />
</template>

<script>
export default {
     methods: {
          handleEditorEvent(type, event) {
               assert(type === 'cursorPosition')
          }
    }
}
</script>
```

**Comment by hzjzx** [comment]
Why not use Vue's event system directly and hide these dirty works inside the library?

**Comment by egoist** [comment]
well they both have pros and cons, in my way, the `js` part get more complicated, in your way the `template` part gets more complicated. I don't mind either of them though. Maybe support the complete list of events then?

**Comment by hzjzx** [comment]
```vue
<template>
  <editor
    @event1="" 
    @event2=""
    @event3=""
    ......
</template>
```
I think this looks better than a huge 'switch' in javascript.
I am going to add all IStandaloneCodeEditor events.

**Comment by egoist** [comment]
Nice, maybe update readme to reflect the changes as well 😀

File: README.md [hunk 1]
```diff
@@ -71,10 +71,32 @@ export default {
...
+| `focus`            | onDidFocusEditor            |                                             |
+| `focusText`        | onDidFocusEditorText        |                                             |
+| `layout`           | onDidLayoutChange           | EditorLayoutInfo                            |
+| `scroll`           | onDidScrollChange           | IScrollEvent                                |
+| `keydown`          | onKeyDown                   | IKeyboardEvent                              |
+| `keyup`            | onKeyUp                     | IKeyboardEvent                              |
+| `mouseDown`        | onMouseDown                 | IEditorMouseEvent                           |
+| `mouseLeave`       | onMouseLeave                | IEditorMouseEvent                           |
+| `mouseMove`        | onMouseMove                 | IEditorMouseEvent                           |
+| `mouseUp`          | onMouseUp                   | IEditorMouseEvent                           |
```
Comment (egoist) [inline README.md]: would be great to add the links of events to monaco-editor's api page?

Comment (hzjzx) [reply README.md]: I dont think it's a good idea now.
The "api page" is generated from a definition file now.
Maybe we can add links after the document of monaco-editor improved.

Comment (egoist) [reply README.md]: maybe simply add a link to https://microsoft.github.io/monaco-editor/api/interfaces/monaco.editor.istandalonecodeeditor.html for now

**Pull request merged**